ui/src/flux/components/SourceDropdown.tsx

```diff
@@ -53,6 +54,11 @@ class SourceDropdown extends PureComponent<Props> {
     onChangeSource(source, type)
   }
 
+  private get widthPixels(): number {
```

Comment:
i'm confused about this. why not just require `width` as a prop? passing in width via a prop is more explicit - someone reading the code can see what the width gets set to and it require someone reading this code to look up how this component defaults its width.

Reply:
So the child component doesn’t require the prop since it conditionally renders based on whether or not it exists

Reply:
So actually, we want it set to 100%, hence the defaulting to 0 (falsey). I had initially set the value as undefined but was getting an error hence the ‘0’ declaration. I can give null a shot - what’re your thoughts?

Reply:
Just rereading what you wrote, sounds good to me!